### DISCUSSION TITLE
/Users/distiller/.gem/ruby/2.7.6/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/app_store_build_number.rb:39:in `get_build_info': \e[31m[!] undefined method `token' for Spaceship::ConnectAPI:Class\e[0m (NoMethodError)

### DISCUSSION THREAD

**Discussion by Karthickveeramanim** [fastlane/fastlane#21230] [Q&A]
Hello There,

We have been using fastlane for the six months for building and deploying android in to the playstore. Now when we are trying to upload iOS using fastlane in circleci we are facing the 'get_build_info' issue. We are following the article which was provided by the circleci for deploying iOS into testflight.

Can anyone please help me to solve this ? we've been stuck with this for nearly two months

**Comment by jkmathew** [comment]
I faced the same issue. In my case, I was trying to call some method on `Spaceship::ConnectAPI` without logging in. 
Try to follow the methods for authentication. https://docs.fastlane.tools/getting-started/ios/authentication/.

If you are using a custom script with `Spaceship::ConnectAPI`, you may have to call login explicitly.
